pallets/validators-manager/src/lib.rs

```diff
@@ -621,6 +623,25 @@ impl<T: Config> Pallet<T> {
         Ok(hex::encode(EthAbiHelper::generate_eth_abi_encoding_for_params_only(&eth_description)))
     }
 
+    pub fn convert_data_to_eth_compatible_encoding_for_collator_registration(
+        action_id: &ActionId<T::AccountId>,
+    ) -> Result<String, DispatchError> {
+        let validators_action_data = Self::try_get_validators_action_data(action_id)?;
+        let eth_description = EthAbiHelper::generate_ethereum_description_for_signature_request(
+            &T::AccountToBytesConvert::into_bytes(&validators_action_data.primary_validator),
+            &validators_action_data.reserved_eth_transaction,
+            validators_action_data.eth_transaction_id,
+        )
+        .map_err(|_| Error::<T>::ErrorGeneratingEthDescription)?;
```

Comment:
This is the bit we want to remove. Instead, create the array directly for the call to `encode(keccak(...))`